crates/e2e/src/setup/onchain_components/safe.rs

```diff
@@ -1,105 +1,101 @@
 use {
     super::{OnchainComponents, TestAccount},
-    contracts::{
-        GnosisSafe,
+    crate::nodes::NODE_HOST,
+    alloy::{
+        primitives::{Address, Bytes, U256},
+        providers::Provider,
+    },
+    contracts::alloy::{
+        GnosisSafe::{self, GnosisSafe::execTransactionCall},
         GnosisSafeCompatibilityFallbackHandler,
         GnosisSafeProxy,
         GnosisSafeProxyFactory,
     },
-    ethcontract::{Bytes, H160, H256, U256},
+    ethcontract::transaction::TransactionBuilder,
+    ethrpc::alloy::{conversions::IntoAlloy, provider_with_account},
     hex_literal::hex,
     model::{
         DomainSeparator,
         order::OrderCreation,
         signature::{Signature, hashed_eip712_message},
     },
     shared::ethrpc::Web3,
-    web3::{
-        signing,
-        signing::{Key, SecretKeyRef},
-    },
+    std::marker::PhantomData,
+    web3::signing::{self},
 };
 
-macro_rules! tx_safe {
-    ($acc:expr_2021, $safe:ident, $call:expr_2021) => {{
-        let call = $call;
-        $crate::tx!(
-            $acc,
-            $safe.exec_transaction(
-                call.tx.to.unwrap(),
-                call.tx.value.unwrap_or_default(),
-                ::ethcontract::Bytes(call.tx.data.unwrap_or_default().0),
-                Default::default(),
-                Default::default(),
-                Default::default(),
-                Default::default(),
-                Default::default(),
-                Default::default(),
-                $crate::setup::safe::gnosis_safe_prevalidated_signature($acc.address()),
-            )
-        );
-    }};
-}
-
 pub struct Infrastructure {
-    pub factory: GnosisSafeProxyFactory,
-    pub fallback: GnosisSafeCompatibilityFallbackHandler,
-    pub singleton: GnosisSafe,
-    web3: Web3,
+    pub factory: GnosisSafeProxyFactory::Instance,
+    pub fallback: GnosisSafeCompatibilityFallbackHandler::Instance,
+    pub singleton: GnosisSafe::Instance,
 }
 
 impl Infrastructure {
-    pub async fn new(web3: &Web3) -> Self {
-        let singleton = GnosisSafe::builder(web3).deploy().await.unwrap();
-        let fallback = GnosisSafeCompatibilityFallbackHandler::builder(web3)
-            .deploy()
+    pub async fn new(signer: TestAccount) -> Self {
+        // Using the NODE_HOST here kind of sucks but this is a workaround
+        // for the alloy provider signing issue
+        let provider = provider_with_account(NODE_HOST, signer.private_key());
```

Comment:
I think this could possibly be worked around by creating a test specific provider with a `NetworkWallet` attached. I believe this should give you signing abilities without having to resort to these workarounds.
Perhaps there are other things as well where having a unique provider setup for tests (as opposed to reusing the same setup we have for the main code) is helpful.

Reply:
I'm not sure I understood your comment but my quick alternative was to create a "test_provider" function that returns one with the default Alloy account as a signer

Reply:
I mean instead of creating new providers ad-hoc in the tests we can also think about replacing the `Web3::new_from_url(NODE_HOST)` call we use to create the 1 provider that is then passed into all the tests.
Although I'm not sure if that's even possible given that we create some PKs only during the test. So the provider that we create there would at least have to have a function where you can register new PKs.
We can continue with your helper function for now but I would still like to think about how this can be made a little nicer.